addon/components/service-worker-update-notify.js

```diff
@@ -27,45 +31,51 @@ export default Component.extend({
   // public
   pollingInterval: 1200000, // 20 minutes in ms
 
-  didInsertElement() {
+  get config() {
+    return getOwner(this).resolveRegistration('config:environment')[configKey];
+  },
+
+  init() {
     this._super(...arguments);
 
-    if (Ember.testing) {
-      this._attachUpdateHandler();
-    } else {
-      // we can't interact with service workers in
-      // the way we want during testing?
-      // seems like it would be complicated if we were
-      // to try.
-      // Also, the tasks use a while(true) loop,
-      // which would prevent tests from every finishing
-      this.setupTask.perform();
+    // NOTE: this.attrs *shouldn't* be used
+    //       as it's poorly documented.
+    //       Consider moving to a glimmer component
+    //       where this.args are available.
+    //
+    // eslint-disable-next-line ember/no-attrs-in-components
+    let wasPassedIn = Boolean(this.attrs.pollingInterval);
```

Comment:
Is `hasOwnProperty()` not working for `this.attrs`? Would be easier to read IMO.

Reply:
Oh, it probably would, I didn't think of that. Thanks!